tlsobs-api/handlers.go

```diff
@@ -44,9 +49,47 @@ func ScanHandler(w http.ResponseWriter, r *http.Request) {
 	db := val.(*pg.DB)
 
 	domain := r.FormValue("target")
-
 	if validateDomain(domain) {
 
+		rescan := false
+		if r.FormValue("rescan") == "true" {
+			rescan = true
+		}
+
+		previd, prevtime, err := db.GetLastScanTimeForTarget(domain)
+		if err != nil {
+			log.WithFields(logrus.Fields{
+				"domain": domain,
+				"error":  err.Error(),
+			}).Error("Could not get last scan for target")
+			err = errors.New("Could not get last scan for target")
+			return
+		}
+
+		if previd != -1 { // check if previous scan exists
+			if time.Now().UTC().Sub(prevtime).Hours() <= scanRefreshRate {
+				if !rescan {
+					// no rescan requested so return previous scan in any case
+					// this includes the rate limiting with no rescan case
+					resp := fmt.Sprintf(`{"scan_id":"%d"}`, previd)
+					w.Header().Set("Content-Type", "application/json")
+					w.WriteHeader(http.StatusOK)
+					fmt.Fprint(w, resp)
+					return
+				}
+
+				// forced rescan has been requested
+				if time.Now().UTC().Sub(prevtime).Minutes() <= 3 { // rate limit scan requests for same target
+					if rescan {
+						w.WriteHeader(429) // 429 http status code is not exported ( https://codereview.appspot.com/7678043/ )
+						fmt.Fprint(w, "")
```

Comment:
What about returning `Last scan scheduled X seconds ago, wait another Y seconds to rescan` ?